transit_odp/timetables/tests/test_dataframes.py

```diff
@@ -0,0 +1,163 @@
+            },
+            {
+                "atco_code": "02901353",
+                "bus_stop_type": "flexible",
+                "journey_pattern_id": "PB0002032:467-jp_1",
+                "service_code": "PB0002032:467",
+                "direction": "outbound",
+            },
+            {
+                "atco_code": "02901278",
+                "bus_stop_type": "flexible",
+                "journey_pattern_id": "UZ000WOCT:216-jp_2",
+                "service_code": "UZ000WOCT:216",
+                "direction": "outbound",
+            },
+        ]
+    )
+    # test
+    stoppoint_df = flexible_journey_patterns_to_dataframe(services)
+    assert check_frame_equal(stoppoint_df, expected_stopoint_df)
+
+
+def test_flexible_stop_points_from_journey_details():
+    # setup
+    flexible_journey_pattern_details = pd.DataFrame(
+        [
+            {
+                "file_id": "339cfb45-3123-4828-8f60-28489f5646cc",
+                "journey_pattern_id": "PF0000508:11-jp_1",
+                "atco_code": "030058880001",
+                "bus_stop_type": "flexible",
+                "service_code": "PF0000508:11",
+                "direction": "outbound",
+            },
+            {
+                "file_id": "339cfb45-3123-4828-8f60-28489f5646cc",
+                "journey_pattern_id": "PF0000508:11-jp_1",
+                "atco_code": "030058860001",
+                "bus_stop_type": "flexible",
+                "service_code": "PF0000508:11",
+                "direction": "outbound",
+            },
+            {
+                "file_id": "339cfb45-3123-4828-8f60-28489f5646cc",
+                "journey_pattern_id": "PF0000508:11-jp_1",
+                "atco_code": "030058920001",
+                "bus_stop_type": "fixed_flexible",
+                "service_code": "PF0000508:11",
+                "direction": "outbound",
+            },
+            {
+                "file_id": "339cfb45-3123-4828-8f60-28489f5646cc",
+                "journey_pattern_id": "PF0000508:11-jp_1",
+                "atco_code": "030058870001",
+                "bus_stop_type": "fixed_flexible",
+                "service_code": "PF0000508:11",
+                "direction": "outbound",
+            },
+        ]
+    )
+
+    expected_stopoint_df = pd.DataFrame(
+        [
+            {"atco_code": "030058880001", "bus_stop_type": "flexible"},
+            {"atco_code": "030058860001", "bus_stop_type": "flexible"},
+            {"atco_code": "030058920001", "bus_stop_type": "fixed_flexible"},
+            {"atco_code": "030058870001", "bus_stop_type": "fixed_flexible"},
+        ]
+    ).set_index("atco_code")
+
+    # test
+    stop_points = flexible_stop_points_from_journey_details(
+        flexible_journey_pattern_details
+    )
+    assert check_frame_equal(stop_points, expected_stopoint_df)
+
+
+def test_flexible_jp_from_journey_details():
+    # setup
+    flexible_journey_pattern_details = pd.DataFrame(
```

Comment:
Could we update the last element with a different jp_id or add another test case with more than one journey patterns